Client/Ecosia/Extensions/HomepageViewController+Ecosia.swift

```diff
@@ -7,38 +7,24 @@ import Core
 
 protocol HomepageViewControllerDelegate: AnyObject {
     func homeDidTapSearchButton(_ home: HomepageViewController)
-    func homeDidPressPersonalCounter(_ home: HomepageViewController, completion: (() -> Void)?)
+    func showSettingsWithDeeplink(to destination: AppSettingsDeeplinkOption)
 }
 
-extension HomepageViewController {
-    func configureEcosiaSetup() {
-        personalCounter.subscribe(self) { [weak self] _ in
-            guard let self = self else { return }
-            self.updateTreesCell()
-        }
-
-        referrals.subscribe(self) { [weak self] _ in
-            guard let self = self else { return }
-            self.updateTreesCell()
-        }
-    }
-
-    func updateTreesCell() {
-        guard let impactCell = viewModel.impactViewModel.cell else { return }
-        impactCell.display(treesCellModel, animated: false)
-    }
+protocol SharedHomepageCellDelegate: AnyObject {
+    func openLink(url: URL)
+    func invalidateLayout(at indexPaths: [IndexPath])
```

Comment:
The reason for the `invalidateLayout` method is to force the about cells to update their sizes when a cell is expanded. To do that we actually need access to `HomepageViewController.collectionView`, so that's why I created the delegate method to be executed at the VC level. This was initially part of an about section specific delegate, but decided to move to this generic one since it is a generic logic that other cells might need.

To tour point, do you mean that you think those should be two separate delegate protocols? I put them in the same one since they are implemented by the same class, but I understand your reasoning given not all cells will need both methods. There would only be the consequence, for example on the about section case, that some cells would require both delegates then which I guess is not a problem.

Reply:
Will separate it then!